crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -230,7 +243,7 @@ impl BlobVec {
     /// The `value` must match the [`layout`](`BlobVec::layout`) of the elements in the [`BlobVec`].
     #[inline]
     pub unsafe fn push(&mut self, value: OwningPtr<'_>) {
-        self.reserve_exact(1);
+        self.reserve(1);
         let index = self.len;
         self.len += 1;
```

Comment:
Unrelated to your fix, but the issue with integer overflow for ZST is still there:

https://github.com/bevyengine/bevy/pull/10799